frame/treasury/src/lib.rs

```diff
@@ -294,41 +289,39 @@ pub mod pallet {
 		},
 		/// The inactive funds of the pallet have been updated.
 		UpdatedInactive { reactivated: BalanceOf<T, I>, deactivated: BalanceOf<T, I> },
+		/// The proposal was paid successfully
+		ProposalPaymentSuccess {
+			proposal_index: ProposalIndex,
+			asset_id: T::AssetId,
+			amount: BalanceOf<T, I>,
+		},
+		// The proposal payment failed. Payment will be retried in next spend period.
+		ProposalPaymentFailure {
+			proposal_index: ProposalIndex,
+			asset_id: T::AssetId,
+			amount: BalanceOf<T, I>,
+		},
 	}
 
 	/// Error for the treasury pallet.
 	#[pallet::error]
 	pub enum Error<T, I = ()> {
-		/// Proposer's balance is too low.
-		InsufficientProposersBalance,
 		/// No proposal or bounty at that index.
 		InvalidIndex,
-		/// Too many approvals in the queue.
-		TooManyApprovals,
+		// /// Too many approvals in the queue.
+		// TooManyApprovals,
 		/// The spend origin is valid but the amount it is allowed to spend is lower than the
 		/// amount to be spent.
 		InsufficientPermission,
-		/// Proposal has not been approved.
-		ProposalNotApproved,
+		/// Unable to convert asset to native balance
+		BalanceConversionFailed,
```

Comment:
We could also bubble up [the error returned by ](https://github.com/paritytech/substrate/pull/13608/files#diff-d3f111178cf5e2e266b3f267b8c6a43f808adb4237960f0241850d756b749236R177)`BalanceConversion`.

Reply:
Yeah, I think i'm doing that in the PR. Can you take a look?